tests/io/test_multi_table.py

```diff
@@ -0,0 +1,271 @@
+            )
+        test_sdata = SpatialData(
+            shapes={"poly": test_shapes["poly"], "multipoly": test_shapes["multipoly"]},
+            table={"poly_annotate": table, "multipoly_annotate": table2},
+        )
+        test_sdata.write(tmpdir)
+        test_sdata = SpatialData.read(tmpdir)
+        assert len(test_sdata.tables) == 2
+
+    def test_single_table_multiple_elements(self, tmp_path: str):
+        tmpdir = Path(tmp_path) / "tmp.zarr"
+        table = _get_table(region=["poly", "multipoly"])
+        subset = table[table.obs.region == "multipoly"]
+        with pytest.raises(ValueError, match="Regions in"):
+            TableModel().validate(subset)
+
+        test_sdata = SpatialData(
+            shapes={
+                "poly": test_shapes["poly"],
+                "multipoly": test_shapes["multipoly"],
+            },
+            table=table,
+        )
+        test_sdata.write(tmpdir)
+        SpatialData.read(tmpdir)
+
+        # # use case example 1
+        # # sorting the shapes visium0 to match the order of the table
+        # sdata["visium0"][sdata.table.obs["__instance_id__"][sdata.table.obs["__spatial_element__"] == "visium0"]]
+        # assert ...
+        # # use case example 2
+        # # subsetting and sorting the table to match the order of the shapes visium0
+        # sub_table = sdata.table[sdata.table.obs["__spatial_element"] == "visium0"]
+        # sub_table.set_index(keys=["__instance_id__"])
+        # sub_table.obs[sdata["visium0"]]
+        # assert ...
+
+    def test_multiple_table_without_element(self, tmp_path: str):
+        tmpdir = Path(tmp_path) / "tmp.zarr"
+        table = _get_table(region=None, region_key=None, instance_key=None)
+        table_two = _get_table(region=None, region_key=None, instance_key=None)
+
+        sdata = SpatialData(
+            tables={"table": table, "table_two": table_two},
+        )
+        sdata.write(tmpdir)
+        SpatialData.read(tmpdir)
+
+    def test_multiple_tables_same_element(self, tmp_path: str):
+        tmpdir = Path(tmp_path) / "tmp.zarr"
+        table = _get_table(region="test_shapes")
+        table2 = _get_table(region="test_shapes")
+
+        test_sdata = SpatialData(
+            shapes={
+                "test_shapes": test_shapes["poly"],
+            },
+            tables={"table": table, "table2": table2},
+        )
+        test_sdata.write(tmpdir)
+        SpatialData.read(tmpdir)
+
+
+def test_concatenate_sdata_multitables():
+    sdatas = [
+        SpatialData(
+            shapes={f"poly_{i + 1}": test_shapes["poly"], f"multipoly_{i + 1}": test_shapes["multipoly"]},
+            tables={"table": _get_table(region=f"poly_{i + 1}"), "table2": _get_table(region=f"multipoly_{i + 1}")},
+        )
+        for i in range(3)
+    ]
+
+    with pytest.raises(KeyError, match="Tables must have"):
+        concatenate(sdatas)
+
+    merged_sdata = concatenate(sdatas, concatenate_tables=True)
```

Comment:
Nice! I think here we need to test also the case concatentate_tables=False when the sdata object has > 1 table.

Reply:
That is done at line 238 no? Or do you mean the case of SpatialData objects without table names being the same?

Reply:
I think it's not done. I mean the case in which there are two spatialdata objects with tables with different names and then calling `concatenate()` with `concatenate_tables=False`.

Reply:
For the record, @LucaMarconato and I discussed this last week in a brief call. The behaviour of concatenating is getting quite complex as we have no way a priori to know which tables a user would like to concatenate. This would require another argument taking something as a `list` of `list`. Instead we choose for tables having the same name to have an integer suffix in case `concatenate_tables=False` and for those to be concatenate if set to `True`. A user can afterwards still concatenate those tables the user would like to concatenate.

Reply:
resolved with [475c566](https://github.com/scverse/spatialdata/pull/410/commits/475c5666810871bf1340fd29ddf523cd60111d1a)